crates/driver/src/util/math.rs

```diff
@@ -13,7 +13,16 @@ pub fn mul_ratio(x: U256, q: U256, d: U256) -> Option<U256> {
         return Some(res / d);
     }
 
-    x.full_mul(q).checked_div(d.into())?.try_into().ok()
+    // SAFETY: at this point !d.is_zero() upholds
+    let div = (x.widening_mul(q)) / U512::from(d);
+
+    // After the division, check the most significant limbs (>U256) for overflow
+    let limbs = div.into_limbs();
+    if limbs[4..] != [0, 0, 0, 0] {
```

Comment:
Maybe we can make the intent clearer by writing:
`limbs.iter().skip(4).all(|x| x != 0)` meaning - all most significant bytes above 4th should be zero.
```suggestion
    if limbs.iter().skip(4).all(|x| x != 0) {
```

Reply:
I had `limbs[4..].iter().any(|l| l != 0)` but I feel this is more explicit. In this place it boils down to preference IMO